python/tvm/contrib/hexagon/hexagon_profiler.py

```diff
@@ -0,0 +1,118 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Define HexagonProfiler class to enable profiling for Hexagon"""
+
+import os
+import subprocess
+from tvm.ir.transform import PassContext
+from tvm.contrib.hexagon.profiling.process_lwp_data import process_lwp_output
+from tvm.relay.backend.executor_factory import ExecutorFactoryModule
+from tvm.contrib import utils
+
+
+class HexagonProfiler:
+    """Hexagon Profiler"""
+
+    def __init__(self, module: ExecutorFactoryModule, hexagon_server_process, enable_debug):
+        """Configure HexagonProfiler"""
+        # Save test .so to process profiling data
+        dso_binary = "test_binary.so"
```

Comment:
I think this needs to be abstracted to allow for other binary names

Reply:
That's a good suggestion, @joshherr-quic. Thanks!